recipes/conversation_view/attachment_carousel/media_components/image_carousel.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <div
+    role="presentation"
+    class="d-ps-relative"
+    @focusin="showCloseButton = true"
+    @focusout="showCloseButton = false"
+    @mouseenter="showCloseButton = true"
+    @mouseleave="showCloseButton = false"
+  >
+    <dt-image-viewer
+      image-button-class="d-h64 d-w64 d-bar4 d-ba d-baw6 d-bc-subtle"
+      :image-src="mediaItem.path"
+      :image-alt="mediaItem.altText"
+      :close-aria-label="closeAriaLabel"
+      :aria-label="clickToOpenAriaLabel"
+    />
+
+    <!-- Loader / Close button -->
+    <div
+      class="dt-image-carousel--top-right"
+    >
+      <dt-progress-bar
+        v-if="mediaItem.isUploading && !showCloseButton"
+        class="dt-image-carousel--progress-bar d-ba d-bc-subtle d-baw2"
+        :progress="mediaItem.progress"
+      />
+      <dt-button
+        v-else
+        v-show="showCloseButton"
```

Comment:
bit confused with the v-else / v-show thing we got goin on here. Can we not just v-else? what are we working around here?

Reply:
on hover of the image, we have to show close button irrespective of if its uploading. 

The close button will be shown at all times but will be hidden if its not hovered. The uploading svg will only come up if that isUploading prop is set.

Reply:
so

```
<dt-progress-bar v-if="mediaItem.isUploading && !showCloseButton />
<dt-button v-else-if="showCloseButton" />
```